lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2025,10 +2028,14 @@ Attribute DotOperandEncodingAttr::parse(AsmParser &parser, Type type) {
 }
 
 void DotOperandEncodingAttr::print(mlir::AsmPrinter &printer) const {
-  auto mmaParent = getParent().dyn_cast<NvidiaMmaEncodingAttr>();
+  auto parent = getParent();
+  auto mmaParent = parent.dyn_cast<NvidiaMmaEncodingAttr>();
   printer << "<{"
           << "opIdx = " << getOpIdx() << ", parent = " << getParent();
-  if (mmaParent && mmaParent.isAmpere())
+  bool isAmdWithKWidth =
+      parent.isa<AMDMfmaEncodingAttr>() || parent.isa<AMDWmmaEncodingAttr>();
```

Comment:
This seems like a similar inconsistency to Ampere/Hopper.  Here we're printing kWidth if AMDWmma, but if AMDWmma we ignore any parsed value and unconditionally do `kWidth = AMDWmmaEncodingAttr::getMNKDimPerWMMAInstr()[2];`